tests/analyzer/sourcecode/download-executable.py

```diff
@@ -70,4 +70,6 @@
 open(executable2, "wb").write(response.content)
 os.system(f'set __COMPACT_LAYER=RunAsInvoker | start +x {executable2}')
 
-
+# ruleid: download-executable
```

Comment:
This test isn't passing because this rule only matches when the download is followed by writing the file to disk and chmod'ing it manually. We could probably tweak it to make the chmod'ing optional, and consider that `urlretrieve` is similar to manually writing to disk, wdyt?

Reply:
@jxdv any thoughts? Still happy to merge your contribution

Reply:
Hi @christophetd, sorry I forgot about this PR :)

> This test isn't passing because this rule only matches when the download is followed by writing the file to disk and chmod'ing it manually. We could probably tweak it to make the chmod'ing optional, and consider that `urlretrieve` is similar to manually writing to disk, wdyt?

I understand what you are saying, but I am not sure how to apply the suggestion you've made. Could you provide a little more detail?

Reply:
Hello team I was looking at this, 
I think that this is not in fact taint analysis, but a standard search (I see no sink function having data being tainted, in fact, arguments are hardcoded). 

I was able to do it using taint mode anyways to avoid having 2 rules for the same detection kind but I made another [branch / PR](https://github.com/DataDog/guarddog/pull/328) to don't mess with this one